plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/EqualityDeleteFilter.java

```diff
@@ -17,51 +17,227 @@
+        {
+            synchronized (lockToken) {
+                return dataVersion < deleteDataSequenceNumber || (!isLoaded && dataVersion == deleteDataSequenceNumber);
+            }
+        }
+
+        public boolean acquireLoading()
+        {
+            synchronized (lockToken) {
+                if (isLoading) {
+                    return false;
+                }
+                else {
+                    isLoading = true;
+                    return true;
+                }
+            }
+        }
+
+        public void setDataVersion(long newDataVersion)
+        {
+            synchronized (lockToken) {
+                this.dataVersion = newDataVersion;
+            }
+        }
+
+        public void setLoaded()
+        {
+            synchronized (lockToken) {
+                this.isLoaded = true;
+            }
+        }
+
+        public void releaseLoading()
+        {
+            synchronized (lockToken) {
+                isLoading = false;
+            }
+        }
+    }
+
+    private final Schema deleteSchema;
+    private final StructLikeMap<DataSequenceNumber> deleteMap;
+    private final ReentrantReadWriteLock readWriteLock;
+    private final HashMap<String, FileToken> loadedFiles;
+
+    public EqualityDeleteFilter(Schema deleteSchema)
+    {
+        this.deleteSchema = requireNonNull(deleteSchema, "deleteSchema is null");
+        this.deleteMap = StructLikeMap.create(deleteSchema.asStruct());
+        this.loadedFiles = new HashMap<>();
+        this.readWriteLock = new ReentrantReadWriteLock();
+    }
+
+    public FileToken getLoadFileToken(String fileName, long deleteFileSequenceNumber)
+    {
+        synchronized (loadedFiles) {
+            return loadedFiles.computeIfAbsent(fileName, k -> new FileToken(deleteFileSequenceNumber));
+        }
     }
 
     @Override
-    public RowPredicate createPredicate(List<IcebergColumnHandle> columns)
+    public RowPredicate createPredicate(List<IcebergColumnHandle> columns, long splitDataSequenceNumber)
     {
         Type[] types = columns.stream()
                 .map(IcebergColumnHandle::getType)
                 .toArray(Type[]::new);
 
-        return (page, position) -> {
-            StructLike row = new LazyTrinoRow(types, page, position);
-            return !deletedRows.test(row);
-        };
+        Schema fileSchema = schemaFromHandles(columns);
+        boolean hasRequiredColumns = columns.size() >= fileSchema.columns().size();
+        for (Types.NestedField column : deleteSchema.columns()) {
+            hasRequiredColumns = hasRequiredColumns && fileSchema.findField(column.fieldId()) != null;
+        }
+        if (!hasRequiredColumns) {
+            // If we don't have all the required columns this delete filter can't be applied.
+            // The iceberg split manager is responsible for making sure that we have all the required columns when a delete filter should be applied
+            return (page, position) -> true;
```

Comment:
Should this throw an exception instead of returning an always-true predicate? It seems like the comment is describing a scenario that should only happen in the presence of a bug